tests/unit/commands/samconfig/test_samconfig.py

```diff
@@ -0,0 +1,483 @@
+"""
+Tests whether SAM Config is being read by all CLI commands
+"""
+
+import os
+import shutil
+import tempfile
+from pathlib import Path
+from contextlib import contextmanager
+from samcli.lib.config.samconfig import SamConfig, DEFAULT_ENV
+
+from click.testing import CliRunner
+
+from unittest import TestCase
+from mock import patch, ANY
+import logging
+
+LOG = logging.getLogger()
+logging.basicConfig()
+
+
+class TestSamConfigForAllCommands(TestCase):
+    def setUp(self):
+        self._old_cwd = os.getcwd()
+
+        self.scratch_dir = tempfile.mkdtemp()
+        Path(self.scratch_dir, "envvar.json").write_text("{}")
+
+        os.chdir(self.scratch_dir)
+
+    def tearDown(self):
+        os.chdir(self._old_cwd)
+        shutil.rmtree(self.scratch_dir)
+        self.scratch_dir = None
+
+    @patch("samcli.commands.init.do_cli")
+    def test_init(self, do_cli_mock):
+        config_values = {
+            "no_interactive": True,
+            "location": "github.com",
+            "runtime": "nodejs10.x",
+            "dependency_manager": "maven",
+            "output_dir": "myoutput",
+            "name": "myname",
+            "app_template": "apptemplate",
+            "no_input": True,
+            "extra_context": '{"key": "value", "key2": "value2"}',
+        }
+
+        with samconfig_parameters(["init"], self.scratch_dir, **config_values) as config_path:
+            from samcli.commands.init import cli
+
+            LOG.debug(Path(config_path).read_text())
+            runner = CliRunner()
+            result = runner.invoke(cli, [])
+
+            LOG.info(result.output)
+            LOG.info(result.exception)
+            if result.exception:
+                LOG.exception("Command failed", exc_info=result.exc_info)
+            self.assertIsNone(result.exception)
+
+            do_cli_mock.assert_called_with(
```

Comment:
Nit: can we make `assert_called_with(a=b,c=d,e=f,g=h)` instead of `assert_called_with(b,d,f,h)`. That way its explicit.

Reply:
I wanted to verify the ordering as well, because that's how the do_cli method is called. Makes sense?